layouts/shortcodes/api-reference.html

```diff
@@ -0,0 +1,3 @@
+{{ $kind := .Get "kind" }}
+{{ $apiVersion := .Get "apiVersion" }}
```

Comment:
@feloy, In this shortcode, would you read the data in the front matter of the individual resource pages and create links from this data or insert the actual body of the resource page?

Reply:
@kbhawkey , I would like to create a link to the kind/apiversion page, from the info on the front matter of this same destination page.

Reply:
Here is one way:
testing this shortcode, link-definition.html, generates a link. I did not hook up the path; you could generate the path.

```
{{ $op_name := .Get "name" }}
{{ $op_path := .Get "path" }}
{{ $id := "deployment-test" }}
{{- $opBundle := site.GetPage "page" "docs/reference/kubernetes-api/deployment" -}}
{{- $opItems :=  $opBundle.Resources.ByType "page" -}}
{{- $term_info := $opItems.GetMatch (printf "%s.md" $id ) -}}
{{ $group := $term_info.Params.group }}
{{ $metadata := $term_info.Params.api_metadata }}
<a href="">{{ $group }} {{ $metadata.kind }}</a>
```

I created a test file as "deployment-test.md (headless bundle) in the deployment directory". Use the shortcode in some other file.

```
---
api_metadata:
  kind: "Deployment"
  apiVersion: "apps/v1"
  import: "k8s.io/api/apps/v1"
title: "Deployment-Test"
description: "Deployment enables declarative updates for Pods and ReplicaSets."
draft: false
id: deployment-id
collapsible: false
group: poohbah
weight: 7
content_type: concept
# deployment-test.md
---

### Deployment

`apiVersion: apps/v1`

`import "k8s.io/api/apps/v1"`

Deployment enables declarative updates for Pods and ReplicaSets.
```

Reply:
In my test, I made deployment-test a page in a headless bundle. The page is not published.
You could point the link to the real page for a resource -- perhaps a page that uses a corresponding bundle page.

Reply:
Cool - that shortcode could help with https://github.com/kubernetes/website/issues/24441 too.